libraries/AC_AutoTune/AC_AutoTune_Heli.cpp

```diff
@@ -706,6 +706,31 @@ void AC_AutoTune_Heli::save_tuning_gains()
     reset();
 }
 
+// report final gains for a given axis to GCS
+void AC_AutoTune_Heli::report_final_gains(AxisType test_axis) const
+{
+    switch (test_axis) {
+        case ROLL:
+            report_axis_gains("Roll", tune_roll_rp, tune_roll_rff*AUTOTUNE_FFI_RATIO_FINAL, tune_roll_rd, tune_roll_rff, tune_roll_sp, tune_roll_accel);
+            break;
+        case PITCH:
+            report_axis_gains("Pitch", tune_pitch_rp, tune_pitch_rff*AUTOTUNE_FFI_RATIO_FINAL, tune_pitch_rd, tune_pitch_rff, tune_pitch_sp, tune_pitch_accel);
+            break;
+        case YAW:
+            report_axis_gains("Yaw", tune_yaw_rp, tune_yaw_rp*AUTOTUNE_YAW_PI_RATIO_FINAL, tune_yaw_rd, tune_yaw_rff, tune_yaw_sp, tune_yaw_accel);
+            break;
+    }
+}
+
+// report gain formating helper
+void AC_AutoTune_Heli::report_axis_gains(const char* axis_string, float rate_P, float rate_I, float rate_D, float rate_ff, float angle_P, float max_rate) const
+{
+    gcs().send_text(MAV_SEVERITY_NOTICE,"AutoTune: %s complete", axis_string);
+    gcs().send_text(MAV_SEVERITY_NOTICE,"AutoTune: %s Rate: P:%0.3f, I:%0.3f, D:%0.4f, FF:%0.3f",axis_string,rate_P,rate_I,rate_D,rate_ff);
```

Comment:
Heli’s will need 5 digits after the decimal point for the D gain and use 4 digits for the P gain.

Reply:
done